Python/ceval.c

```diff
@@ -4637,40 +4643,48 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
 
             // Check if the call can be inlined or not
             PyObject *function = PEEK(oparg + 1);
-            int inline_call = 0;
             if (Py_TYPE(function) == &PyFunction_Type) {
                 PyCodeObject *code = (PyCodeObject*)PyFunction_GET_CODE(function);
-                int is_coro = code->co_flags & (CO_GENERATOR | CO_COROUTINE | CO_ASYNC_GENERATOR);
-                inline_call = (is_coro || cframe.use_tracing) ? 0 : 1;
+                STACK_SHRINK(oparg + 1);
```

Comment:
I am fine with this but I find this a bit more difficult to read than the previous code. This has two conditionals nested and is not immediately clear what's the unoptimized path

Reply:
I don't think there is an optimized or unoptimized path here. The specialized versions can worry about fast paths.